models/template_helpers.go

```diff
@@ -0,0 +1,222 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"time"
+
+	"github.com/percona-platform/saas/pkg/alert"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+)
+
+func checkUniqueTemplateName(q *reform.Querier, name string) error {
+	if name == "" {
+		panic("empty template name")
```

Comment:
why do you choose to panic here instead of returing an error?

Reply:
Same as in https://github.com/percona/pmm-managed/blob/c2cd5d57da2f4dccfe7b2694e1f75e0f53125ebd/models/agent_helpers.go#L33

Reply:
I personally prefer not to panic) But it's common in `checkXXX` methods here.

Reply:
Panic there is used to indicate programmer's error – that function should not be called with empty ID; it is caller's responsibility to check that ID is not empty if it comes from the API call, etc. (in practice, that should be covered by protobuf validators, but please check that)